src/web/demo.rkt

```diff
@@ -443,13 +419,33 @@
 (define (improve req)
   (improve-common
    req
-   (λ (hash formula)
-     (unless (already-computed? hash formula)
-       (semaphore-wait (run-xxx hash formula 'improve (list))))
-
-     (redirect-to (add-prefix (format "~a.~a/graph.html" hash *herbie-commit*)) see-other))
+   (λ (formula)
+    (thread-improve-and-save formula)
+    (redirect-to (add-prefix (format "~a.~a/graph.html" hash *herbie-commit*)) see-other))
    (url main)))
 
+;; Bad name but encapslates shared work of runing the improve command
+;; saving the results to disk
+(define (thread-improve-and-save formula)
+  ;; Should seed optinal be passed in through the req from the UI?
+  (define seed get-seed)
```

Comment:
```suggestion
  (define seed (get-seed))
```

That you didn't catch this tells me this is either dead code or not tested. Both are bad.